### DISCUSSION TITLE
delete the user account

### DISCUSSION THREAD

**Discussion by mshannaq** [codeigniter4/shield#975] [Q&A] [answered]
Hi,

I'm wondering if there is a method available to delete a user account using CodeIgniter Shield, or is it necessary to handle it manually?

I couldn't locate any specific method for deleting a user in the `CodeIgniter\Shield\Entities\User` Entity.

When deleting a username, it's crucial to also remove any associated data for the user from Shield's MySQL tables, such as the `auth_groups_users` table and `auth_identities` table.

Is there a built-in method within Shield? or is manual deletion the recommended approach?

**Comment by kenjis** [comment]
See https://codeigniter4.github.io/shield/user_management/managing_users/#deleting-users

**Comment by mshannaq** [comment, marked as answer]
Sorry I find the solution Shield Docs https://codeigniter4.github.io/shield/user_management/managing_users/

using 
```
// Get the User Provider (UserModel by default)
$users = auth()->getProvider();

$users->delete($user->id, true);
```
